modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -44,13 +42,28 @@ validParams<SingleVariableReturnMappingSolution>()
                         "algorithm. Also use same old defaults for relative_tolerance, "
                         "absolute_tolerance, and max_its.");
 
+  // diagnostic output parameters
+  MooseEnum debug_level_enum("none error always", "none");
+  params.addParam<MooseEnum>(
+      "debug_level", debug_level_enum, "When to output Newton solve information");
+  params.addParam<bool>(
+      "output_iteration_info", false, "Set true to output full Newton iteration history");
+  params.addDeprecatedParam<bool>(
+      "output_iteration_info_on_error",
+      false,
+      "Set true to output Newton iteration information when those iterations fail",
+      "This parameter is replaced by the 'debug_level' parameter");
+  params.addParamNamesToGroup("debug_level output_iteration_info output_iteration_info_on_error",
+                              "Debug");
+
   return params;
 }
 
 SingleVariableReturnMappingSolution::SingleVariableReturnMappingSolution(
     const InputParameters & parameters)
   : _legacy_return_mapping(false),
     _check_range(false),
+    _debug_level(parameters.get<MooseEnum>("debug_level").getEnum<DebugLevel>()),
```

Comment:
Isn't it best practices to set _iteration in the constructor, even if it'll change later?

Reply:
I think the jury is still out on that one, but I have no issue adding the initializer anyways.